### PR TITLE
JNG-6038 Guice rest injector

### PR DESCRIPTION
**Pull request by robertcsakany**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-6038" title="JNG-6038" target="_blank">StoryJNG-6038</a>  Inject Cxf / Rest to Guice / Spring
  </td></table>
  <br />

### CODE REVIEW

File: judo-runtime-core-guice-cxf/src/main/java/hu/blackbelt/judo/runtime/core/jaxrs/cxf/server/guice/providers/CxfJaxrsServerProvider.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+    @Inject(optional = true)
+    @CxfQualifiers.Providers
+    @Nullable
+    private Set<Object> providers = new HashSet<>();
+
+    @Override
+    public ServerHolder get() {
+        ServerHolder serverHolder = new ServerHolder();
+//        JettyContainer jettyContainer = new JettyContainer(cxfJaxRsServerPort, "/");
```
Comment (bencegelei) [inline judo-runtime-core-guice-cxf/src/main/java/hu/blackbelt/judo/runtime/core/jaxrs/cxf/server/guice/providers/CxfJaxrsServerProvider.java]: commented code?

File: judo-runtime-core-guice-cxf/src/main/java/hu/blackbelt/judo/runtime/core/jaxrs/cxf/server/guice/providers/ExtendedObjectMapperProvider.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import com.fasterxml.jackson.module.paramnames.ParameterNamesModule;
+import com.google.inject.Provider;
+
+public class ExtendedObjectMapperProvider implements Provider<ObjectMapper> {
+
+    public static ObjectMapper getExtendedObjectMapper() {
+        return new ObjectMapper()
+                .findAndRegisterModules()
+                .registerModule(new ParameterNamesModule())
+                .registerModule(new Jdk8Module())
```
Comment (bencegelei) [inline judo-runtime-core-guice-cxf/src/main/java/hu/blackbelt/judo/runtime/core/jaxrs/cxf/server/guice/providers/ExtendedObjectMapperProvider.java:38]: Jdk8?

File: judo-runtime-core-guice-cxf/src/test/java/hu/blackbelt/judo/runtime/core/guice/JudoCxfModuleTest.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    int port;
+    String restUri;
+
+    @Inject
+    CxfJaxrsServerProvider.ServerHolder serverHolder;
+
+    @SuppressWarnings({ "rawtypes", "resource" })
+    @BeforeEach
+    void init() throws Exception {
+        Stopwatch timer = Stopwatch.createStarted();
```
Comment (bencegelei) [inline judo-runtime-core-guice-cxf/src/test/java/hu/blackbelt/judo/runtime/core/guice/JudoCxfModuleTest.java:55]: Stopwatch? :DD

File: judo-runtime-core-guice-keycloak/src/main/java/hu/blackbelt/judo/runtime/core/security/keycloak/guice/providers/KeycloakRealmSynchronizerProvider.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+    @Inject
+    JudoModelLoader judoModelLoader;
+
+    public KeycloakRealmSynchronizer get() {
+        Collection<String> corsAllowOrigin;
+        if (this.corsAllowOrigin != null) {
+            corsAllowOrigin = Arrays.stream(this.corsAllowOrigin.split(",")).map(v -> v.trim()).collect(Collectors.toList());
+        } else {
+            corsAllowOrigin = Collections.emptySet();
```
Comment (bencegelei) [inline judo-runtime-core-guice-keycloak/src/main/java/hu/blackbelt/judo/runtime/core/security/keycloak/guice/providers/KeycloakRealmSynchronizerProvider.java]: `Collections.emptySet()` returns an immutable set while `Collectors.toList()` is mutable.

File: judo-runtime-core-guice-keycloak/src/main/java/hu/blackbelt/judo/runtime/core/security/keycloak/guice/providers/KeycloakUserManagerProvider.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    PasswordPolicy defaultPasswordPolicy;
+
+    @Inject
+    TransformationTraceService transformationTraceService;
+
+    @Inject
+    KeycloakAdminClient keycloakAdminClient;
+
+
+    /*
```
Comment (bencegelei) [inline judo-runtime-core-guice-keycloak/src/main/java/hu/blackbelt/judo/runtime/core/security/keycloak/guice/providers/KeycloakUserManagerProvider.java]: commented code?

File: judo-runtime-core-guice-postgresql/src/main/java/hu/blackbelt/judo/runtime/core/guice/dao/rdbms/postgresql/PostgresqlDataSourceProvider.java [hunk 6]
```diff
@@ -22,52 +22,63 @@
...
 
     @Override
     public DataSource get() {
 
+        boolean hikari = true;
+        boolean pooled = true;
+
+        String jdbcUrl = "jdbc:postgresql://" + host + ":" + port + "/" + databaseName;
+        DataSource pg;
+//        if (pooled) {
```
Comment (bencegelei) [inline judo-runtime-core-guice-postgresql/src/main/java/hu/blackbelt/judo/runtime/core/guice/dao/rdbms/postgresql/PostgresqlDataSourceProvider.java]: commented code?

File: judo-runtime-core-guice-postgresql/src/test/java/hu/blackbelt/judo/runtime/core/guice/postgresql/JudoDefaultPostgresqlModuleTest.java [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+        sqlContainer =
+                (PostgreSQLContainer) new PostgreSQLContainer("postgres:16-alpine")
+                        .withDatabaseName("test")
+                        .withUsername("test")
+                        .withDatabaseName("test")
+                        .withExposedPorts(port)
+                        .withAccessToHost(true)
+                        .withStartupTimeout(Duration.ofSeconds(10));
+        sqlContainer.setPortBindings(Arrays.asList(port + ":" + 5432));
```
Comment (bencegelei) [inline judo-runtime-core-guice-postgresql/src/test/java/hu/blackbelt/judo/runtime/core/guice/postgresql/JudoDefaultPostgresqlModuleTest.java:94]: > 5432

shouldn't this number be extracted as a parameter?

File: judo-runtime-core-guice/src/main/java/hu/blackbelt/judo/runtime/core/guice/JudoModelLoader.java [hunk 8]
```diff
@@ -107,14 +154,14 @@ public static JudoModelLoader loadFromDirectory(String modelName, File directory
...
     }
 
-    public static JudoModelLoader loadFromURL(String modelName, URI uri, Dialect dialect) throws Exception {
-        return loadFromURL(modelName, uri, dialect, true);
+    public static JudoModelLoader loadFromURL(String modelName, URI uri, Dialect dialect, boolean loaKeucloak) throws Exception {
+        return loadFromURL(modelName, uri, dialect, true, loaKeucloak);
     }
 
-    public static JudoModelLoader loadFromURL(String modelName, URI uri, Dialect dialect, boolean validate) throws Exception {
+    public static JudoModelLoader loadFromURL(String modelName, URI uri, Dialect dialect, boolean validate, boolean loadKetkloak) throws Exception {
```
Comment (gaborflorian) [inline judo-runtime-core-guice/src/main/java/hu/blackbelt/judo/runtime/core/guice/JudoModelLoader.java]: loaKeycloak, loaKeucloak, loadKetkloak -> I guess this should be loadKeycloak :D

**Review by gaborflorian — APPROVED** [review]

File: judo-runtime-core-guice-cxf/src/test/java/hu/blackbelt/judo/runtime/core/guice/JudoCxfModuleTest.java [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+    public void testAddCall() {
+        int a = 122;
+        int b = 34;
+        WebClient plainAddClient = WebClient.create(restUri);
+        plainAddClient.path("calc/add").path(a + "/" + b).accept("application/json");
+        String res = plainAddClient.get(String.class);
+        assertEquals("156.0", res);
+    }
+
+    /*
```
Comment (Madmat8) [inline judo-runtime-core-guice-cxf/src/test/java/hu/blackbelt/judo/runtime/core/guice/JudoCxfModuleTest.java]: Commented code

**Review by Madmat8 — APPROVED** [review]

File: judo-runtime-core-guice-cxf/src/main/java/hu/blackbelt/judo/runtime/core/jaxrs/cxf/server/guice/providers/ExtendedObjectMapperProvider.java (hunk 2 above)
Comment (robertcsakany) [reply judo-runtime-core-guice-cxf/src/main/java/hu/blackbelt/judo/runtime/core/jaxrs/cxf/server/guice/providers/ExtendedObjectMapperProvider.java:38]: That the name for java temoral type

File: judo-runtime-core-guice-keycloak/src/main/java/hu/blackbelt/judo/runtime/core/security/keycloak/guice/JudoKeycloakModuleConfiguration.java [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+    String keycloakAdminUser = "admin";
+
+    @Builder.Default
+    String keycloakAdminPassword = "admin";
+
+    @Builder.Default
+    String keycloakClientSecret = "";
+
+    @Builder.Default
+    Boolean keycloakRealmSynchronizerSupportLoginByEmail = true;;
```
Comment (noherczeg) [inline judo-runtime-core-guice-keycloak/src/main/java/hu/blackbelt/judo/runtime/core/security/keycloak/guice/JudoKeycloakModuleConfiguration.java]: ;;

File: judo-runtime-core-guice-keycloak/src/main/java/hu/blackbelt/judo/runtime/core/security/keycloak/guice/JudoKeycloakModuleConfiguration.java [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+    String keycloakServerUrl = "http://localhost:8080/auth";
+
+    @Builder.Default
+    String keycloakPublicUrl = "http://localhost:8080/auth";
+
+    @Builder.Default
+    String keycloakAdminUser = "admin";
+
+    @Builder.Default
+    String keycloakAdminPassword = "admin";
```
Comment (noherczeg) [inline judo-runtime-core-guice-keycloak/src/main/java/hu/blackbelt/judo/runtime/core/security/keycloak/guice/JudoKeycloakModuleConfiguration.java]: admin?

File: judo-runtime-core-guice-keycloak/src/main/java/hu/blackbelt/judo/runtime/core/security/keycloak/guice/JudoKeycloakModuleConfiguration.java [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+
+    @Builder.Default
+    String keycloakClientSecret = "";
+
+    @Builder.Default
+    Boolean keycloakRealmSynchronizerSupportLoginByEmail = true;;
+
+    @Builder.Default
+    String keycloakRealmSynchronizerClientAccessTypeForHuman = "CONFIDENTIAL";;
```
Comment (noherczeg) [inline judo-runtime-core-guice-keycloak/src/main/java/hu/blackbelt/judo/runtime/core/security/keycloak/guice/JudoKeycloakModuleConfiguration.java]: ;;

File: judo-runtime-core-guice-keycloak/src/main/java/hu/blackbelt/judo/runtime/core/security/keycloak/guice/JudoKeycloakModuleConfiguration.java [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+    String keycloakRealmSynchronizerCorsAllowOrigin = "";
+
+    @Builder.Default
+    Boolean keycloakRealmSynchronizerAsyncServiceCall = true;
+
+    @Builder.Default
+    Integer keycloakRealmSynchronizerRetryMaxAttempts = 1000;
+
+    @Builder.Default
+    Boolean keycloakRealmSynchronizerRetryExponentialBackoff = true;;
```
Comment (noherczeg) [inline judo-runtime-core-guice-keycloak/src/main/java/hu/blackbelt/judo/runtime/core/security/keycloak/guice/JudoKeycloakModuleConfiguration.java:50]: ;;

File: judo-runtime-core-guice-keycloak/src/main/java/hu/blackbelt/judo/runtime/core/security/keycloak/guice/providers/KeycloakRealmSynchronizerProvider.java [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+
+    @Inject(optional = true)
+    @Named("KeycloakRealmSynchronizerClientAccessTypeForSystem")
+    @Nullable
+    String clientAccessTypeSystem = "BEARER_ONLY";
+
+    @Inject(optional = true)
+    @Named("KeycloakRealmSynchronizerCorsAllowOrigin")
+    @Nullable
+    String corsAllowOrigin;
```
Comment (noherczeg) [inline judo-runtime-core-guice-keycloak/src/main/java/hu/blackbelt/judo/runtime/core/security/keycloak/guice/providers/KeycloakRealmSynchronizerProvider.java]: No default value?

File: judo-runtime-core-guice-keycloak/src/main/java/hu/blackbelt/judo/runtime/core/security/keycloak/guice/JudoKeycloakModuleConfiguration.java (hunk 11 above)
Comment (robertcsakany) [reply judo-runtime-core-guice-keycloak/src/main/java/hu/blackbelt/judo/runtime/core/security/keycloak/guice/JudoKeycloakModuleConfiguration.java]: uh. Ok :)

**Pull request merged**